### PR TITLE
511 group context switching

### PR DESCRIPTION
**Pull request by pospi**
Improved implementation, to supercede #732. Ready to merge if others are happy with this version.

### CODE REVIEW

**Review by tibetsprague — APPROVED** [review]
yummy

File: src/routes/PrimaryLayout/PrimaryLayout.js [hunk 1]
```diff
@@ -248,15 +272,18 @@ const createCommunityRoutes = [
...
   { from: '/p/:postId', to: '/all/p/:postId' },
   { from: '/u/:personId', to: '/m/:personId' },
   { from: '/c/:slug/about', to: '/c/:slug' },
-  { from: '/c/:slug/people', to: '/c/:slug/members' },
-  { from: '/c/:slug/invite', to: '/c/:slug/settings/invite' },
-  { from: '/c/:slug/events', to: '/c/:slug' }
+  { from: '/c/:slug/people', to: '/c/:slug/:view(members)' },
+  { from: '/c/:slug/invite', to: '/c/:slug/:view(settings)/invite' },
+  { from: '/c/:slug/events', to: '/c/:slug' },
+  // redirects for context switching into global contexts
```
Comment (lorenjohnson) [inline src/routes/PrimaryLayout/PrimaryLayout.js:284]: AHHA. It FINALLY got through why this was needed.  :view is obviously not part of the set of things we are clearing on the context switch. Of course. 

Now that I understand what's up better your first solution seems not so bad, but this does feels like a less offensive place to capture (hide perhaps) these exceptions for now--comment included.  Good work~!

File: src/routes/PrimaryLayout/PrimaryLayout.js [hunk 2]
```diff
@@ -248,15 +272,18 @@ const createCommunityRoutes = [
...
+  { from: '/all/members', to: '/all' },
+  { from: '/public/(members|topics)', to: '/public' }
```
Comment (lorenjohnson) [inline src/routes/PrimaryLayout/PrimaryLayout.js:286]: This looked strange to me. No param key needed, I get it, and I'm sure it works as you tested it. It does right?

File: src/routes/AllTopics/AllTopics.js [hunk 3]
```diff
@@ -99,7 +99,7 @@ export default class AllTopics extends Component {
     } = this.props
     const { totalTopicsCached } = this.state
 
-    return <FullPageModal fullWidth>
+    return <FullPageModal fullWidth goToOnClose={baseUrl({ ...routeParams, view: undefined })}>
```
Comment (lorenjohnson) [inline src/routes/AllTopics/AllTopics.js:102]: That looks like a really good solution to that wrinkle 👍

**Review by lorenjohnson — APPROVED** [review]
💯

File: src/routes/PrimaryLayout/PrimaryLayout.js (hunk 2 above)
Comment (pospi) [reply src/routes/PrimaryLayout/PrimaryLayout.js:286]: yep! Seems to land at the intended destinations, and I figured having explicit redirects mapped out there would prevent any future developers from encountering surprises

File: src/routes/AllTopics/AllTopics.js (hunk 3 above)
Comment (pospi) [reply src/routes/AllTopics/AllTopics.js:102]: Heh, it's funny cos I cherry-picked this from my old branch and we didn't like it there but somewhere along the line our relationship to the navigation logic seems to have evolved (;

**Pull request merged**